File: jsimple-io/src/main/java/jsimple/net/Socket.java
package jsimple.net;

import jsimple.io.InputStream;
import jsimple.io.OutputStream;

/**
 * @author Bret Johnson
 * @since 7/24/13 11:07 PM
 */
public abstract class Socket extends jsimple.lang.AutoCloseable {
    public abstract InputStream getInputStream();

    public abstract OutputStream getOutputStream();

    public abstract boolean isClosed();
}

File: jsimple-io/c#/jsimple/net/Socket.cs
namespace jsimple.net
{

	using InputStream = jsimple.io.InputStream;
	using OutputStream = jsimple.io.OutputStream;

	/// <summary>
	/// @author Bret Johnson
	/// @since 7/24/13 11:07 PM
	/// </summary>
	public abstract class Socket : jsimple.lang.AutoCloseable
	{
		public abstract InputStream InputStream {get;}

		public abstract OutputStream OutputStream {get;}

		public abstract bool Closed {get;}
	}

}